grype/version/constraint.go

```diff
@@ -40,3 +40,13 @@ func MustGetConstraint(constStr string, format Format) Constraint {
 	}
 	return constraint
 }
+
+type NonFatalConstraintError struct {
```

Comment:
Requesting change: It would be helpful to other devs if we added a doc comment for this error, explaining why it exists / how it should be used.